worklist/views.py

```diff
@@ -211,3 +184,37 @@ def update_task_info(request):
     results.update(progress=progress, status=status_code, claimed_by=claimed_by)
 
     return JsonResponse({'success': True})
+
+
+# View to display worklists created or being worked upon by the logged-in user
+@login_required
+def show_user_worklists(request):
+    created_worklists = []
+    worked_upon_worklists = []
+    username = request.user.username
+    results = WorkList.objects.filter(created_by=username)
+
+    for result in results:
+        worklist = {
+            'name': result.name,
+            'tags': result.tags,
+            'description': result.description,
+            'psid': result.psid,
+        }
+
+        created_worklists.append(worklist)
+
+    results = Task.objects.filter(claimed_by=username)
+
+    for result in results:
+        worklist = {
+            'name': result.worklist.name,
+            'tags': result.worklist.tags,
+            'description': result.worklist.description,
+            'psid': result.worklist.psid,
+        }
```

Comment:
This is fine, but I don't quite like how we're repeating ourselves: the format of these dicts is expressed twice in this function, once when we load the tasks and once when we load the worklist. It's a trivial case of repetition but I think it makes sense to express it only once.

Can we just pass the worklist objects to the view instead of turning them into dicts?

If that's not possible, how about:

```python
# Get the relevant attributes out of a worklist for the view.
# This could also be { attr : getattr(w, attr) for attr in ('name', 'tags', ...) }
# if you want to be concise, though that might be too hard to read.
worklist_to_dict = lambda w: {'name': w.name, 'tags': w.tags, ...}
created_worklists = map(worklist_to_dict, WorkList.objects.filter(created_by=username))
worked_upon_worklists = map(worklist_to_dict, [r.worklist for r in Task.objects.filter(claimed_by=username)])
```

By the way, [here's a nice read about not repeating knowledge in code](https://www.artima.com/intv/dry.html). I haven't read that interview actually, but their book (The Pragmatic Programmer) was recommended to me by my mentor when I did GSoC and it's really good :)

Reply:
I should have taken care of it beforehand only. My bad. I've made the changes now.
I skimmed through the link. It looks nice. Thanks for the suggestion.